binfmt/binfmt_execmodule.c

```diff
@@ -151,14 +152,22 @@ int exec_module(FAR const struct binary_s *binp,
     }
 
 #if defined(CONFIG_ARCH_ADDRENV) && defined(CONFIG_BUILD_KERNEL)
+  /* Make a copy of the parent's environment here */
+
+  envp = binfmt_copyenv(envp);
+  if (!envp)
+    {
+      ret = -ENOMEM;
+      goto errout_with_args;
+    }
+
   /* Instantiate the address environment containing the user heap */
 
   ret = up_addrenv_select(&binp->addrenv, &oldenv);
   if (ret < 0)
     {
       berr("ERROR: up_addrenv_select() failed: %d\n", ret);
-      binfmt_freeargv(argv);
-      goto errout_with_tcb;
+      goto errout_with_args;
```

Comment:
change errout_with_args to errout_with_envp

Reply:
Sure

Reply:
Done